tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -53,6 +55,27 @@ public TraceContext(IDatadogTracer tracer, TraceTagCollection tags = null)
             get => _samplingPriority;
         }
 
+        public bool IastInitialized
+        {
+            get => _iastRequestContext != null;
+        }
+
+        /// <summary>
+        /// Gets the iast context.
+        /// </summary>
+        internal IastRequestContext IastRequestContext
+        {
+            get
+            {
+                if (_iastRequestContext == null && Iast.Iast.Instance.Settings.Enabled)
+                {
+                    _iastRequestContext = new();
```

Comment:
Could there be a race condition here if accessed from multiple threads?

Reply:
I have added a lock (_syncRoot), reusing the already existing field in the class. I was thinking about creating a specific private lock for this case, but that would also add some more bytes to the tracecontext class...